pkgs/tools/security/openpgp-card-tools/default.nix

```diff
@@ -12,14 +12,17 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "openpgp-card-tools";
-  version = "0.9.3";
+  version = "0.9.4";
 
-  src = fetchCrate {
-    inherit pname version;
-    sha256 = "sha256-F+j8bK0sBBLWlQzLAcvl6BdiI3Dy8ollwTpL7929nJ8=";
+  src = fetchFromGitea {
+    domain = "codeberg.org";
+    owner = "openpgp-card";
+    repo = "openpgp-card-tools";
+    rev = "v${version}";
+    hash = "sha256-ISIABjuh0BC6OUFa5I9Wou+av7Dp4bZH8Aazi6x7cqY=";
   };
```

Comment:
`fetchCrate` doesn't work any more?

Reply:
It still works, but I think it's more preferable to fetch directly from the upstream source. I once discussed with a debian developer who said that "fetching from github is better than from pypi since it's source in its natural form without unknown preprocessing", guess the same applies to crates.

Reply:
Found the original message: https://t.me/archlinuxcn_offtopic/5109887